coalib/bearlib/languages/Language.py

```diff
@@ -366,26 +399,35 @@ def __repr__(self):
         return str(self)
 
     def __gt__(self, other):
-        return limit_versions(self, other, operator.gt)
+        return compare(self, other, operator.gt)
 
     def __lt__(self, other):
-        return limit_versions(self, other, operator.lt)
+        return compare(self, other, operator.lt)
 
     def __ge__(self, other):
-        return limit_versions(self, other, operator.ge)
+        return compare(self, other, operator.ge)
 
     def __le__(self, other):
-        return limit_versions(self, other, operator.le)
+        return compare(self, other, operator.le)
 
     def __eq__(self, other):
-        return limit_versions(self, other, operator.eq)
+        return compare(self, other, operator.eq)
 
     def __ne__(self, other):
-        return limit_versions(self, other, operator.ne)
+        return compare(self, other, operator.ne)
 
     def __or__(self, other):
+        type_equals(self, other)
         return type(self)(*chain(self.versions, other.versions))
 
+    def __and__(self, other):
```

Comment:
Just like `__or__`, it should also work with `other` being a version `str`, `float`, or `int`